src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -13,6 +13,8 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal sealed class SourcePropertySymbol : SourcePropertySymbolBase
     {
+        private readonly bool _containsFieldKeyword;
```

Comment:
>_containsFieldKeyword [](http://example.com/codeflow?start=30&length=21)

The following was said on another thread: "Also we could capture the information per specific accessor and, perhaps, avoid binding an accessor that doesn't use the keyword."

Reply:
@AlekseyTs Makes sense. I'll do that. Since I think there is a lot more work to be done, can we get this merged in a feature branch and continue the rest of the work in follow-up PRs? or if there are specific parts you think are essential in this PR, let me know about them.

Reply:
I think it would be fine marking possible optimizations with clear PROTOTYPE comments precisely explaining what we want to do and then addressing them in follow up PRs. However, I'd like us to address issues about actual APIs behavior in context of this PR. For example, behavior changes around ```IsAutoProperty``` and the like. I think those are fundamental and we should get them right from the start in order to avoid taking dependency on "wrong" behavior as we keep working on the feature.

Reply:
> behavior changes around `IsAutoProperty` and the like.

Can you elaborate more on that?

Reply:
> Can you elaborate more on that?

The relevant threads are already ongoing. It all revolves around when an extra binding of accessors should happen.